Kudu.SiteManagement/SiteManager.cs

```diff
@@ -600,5 +601,66 @@ private ServerManager GetServerManager()
                 }
             }
         }
+
+        public NameValueCollection GetAppSettings(string applicationName)
+        {
+            try
+            {
+                var settingsCollection = new NameValueCollection();
+                using (var iis = GetServerManager())
+                {
+                    var appPool = iis.ApplicationPools[GetAppPool(applicationName)];
+                    foreach (var element in appPool.GetCollection("environmentVariables"))
+                    {
+                        settingsCollection.Add((string)element["name"], (string)element["value"]);
+                    }
+                }
+                return settingsCollection;
+            }
+            catch (Exception)
+            {
+                return null;
+            }
+        }
+
+        public void RemoveAppSetting(string applicationName, string key)
+        {
+            try
+            {
+                using (var iis = GetServerManager())
+                {
+                    var appPool = iis.ApplicationPools[GetAppPool(applicationName)];
+                    var collection = appPool.GetCollection("environmentVariables");
+                    foreach (var element in collection)
+                    {
+                        if ((string)element["name"] == key)
+                        {
+                            collection.Remove(element);
+                            break;
+                        }
+                    }
+                    iis.CommitChanges();
+                }
+            }
+            catch { }
+        }
+
+        public void SetAppSetting(string applicationName, string key, string value)
+        {
+            try
+            {
+                using (var iis = GetServerManager())
+                {
+                    var appPool = iis.ApplicationPools[GetAppPool(applicationName)];
+                    var collection = appPool.GetCollection("environmentVariables");
+                    var addElement = collection.CreateElement("add");
+                    addElement["name"] = key;
+                    addElement["value"] = value;
+                    collection.Add(addElement);
```

Comment:
This blows up if you try to change the value of an existing variable. e.g. it throws:

    Cannot add duplicate collection entry of type 'add' with unique key attribute 'name' set to 'FOO'

Simplest fix is probably to just call `RemoveAppSetting(applicationName, key);` as the first line of the `using`.